src/bundle.ts

```diff
@@ -40,11 +41,16 @@ export const bundleTarget = <T = unknown>(
     workingDocument,
     pointerToPath(bundleRoot),
     pointerToPath(errorsRoot),
+    path,
     keyProvider,
   )(path, { [path]: true }, cur);
 };
 
 const defaultKeyProvider = ({ document, path }: { document: unknown; path: JsonPath }) => {
+  if (path.length === 0) {
+    return 'root';
```

Comment:
Is it possible for this string to collide with anything user-defined?  Why "root"?

Answer: No, we append numbers to reused names.